### PR TITLE
[MSI] Fix heap assert in msi/dialog.c when installing Tortoise GIT

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Purpose

_Fix heap assert in msi/dialog.c._

JIRA issue: [CORE-16693](https://jira.reactos.org/browse/CORE-16693)

## Proposed changes

_Implement Thomas Faber's suggestion to handle heap assert as noted below._
_Since there's not really something sensible to be done in the_
_"path contains no backslashes" case (adding a backslash seems pointless),_
_the PathAddBackslashW call was moved inside the if condition to avoid the crash._

### CODE REVIEW

**Comment by binarymaster** [comment]
Please describe more details in PR description about what's going on here, which would go into the final commit message.

File: dll/win32/msi/dialog.c [hunk 1]
```diff
@@ -2943,8 +2943,16 @@ static UINT msi_dialog_directorylist_up( msi_dialog *dialog )
 
     /* strip off the last directory */
     ptr = PathFindFileNameW( path );
+#if __REACTOS__
```
Comment (HBelusca) [inline dll/win32/msi/dialog.c:2946]:
```suggestion
#ifdef __REACTOS__
```

**Review by HBelusca — COMMENTED** [review]
the 'if' must be 'ifdef'.
I suppose the heap assert is due to the fact it gets corrupted in the case the ptr points to the path, and the unconditional PathAddBackslashW adds a backslash + adds terminating NULL after the end of the allocated buffer (path).

File: dll/win32/msi/dialog.c (hunk 1 above)
Comment (Doug-Lyons) [reply dll/win32/msi/dialog.c]: Fixed. Thanks.

**Review by JoachimHenze — APPROVED** [review]
Approved.
Wine merge-request pending https://gitlab.winehq.org/wine/wine/-/merge_requests/5011

**Review by HBelusca — APPROVED** [review]
OK let's see how it goes on Wine side and if accepted there, merge it (I hope it won't take ages)

**Comment by Doug-Lyons** [comment]
> It's in!

Thanks, @DarkShadow44 !

**Comment by Doug-Lyons** [comment]
> It's in!

@HBelusca, since it's in Wine, should I remove the '#ifdef .../#else' code now?

**Comment by binarymaster** [comment]
Just convert this PR to a Wine cherry-pick of commit https://github.com/wine-mirror/wine/commit/bbce5d014db7f023b133d6d09e6846e027586f7d

**Review by binarymaster — APPROVED** [review]

**Comment by Doug-Lyons** [comment]
Thanks, @binarymaster for your approval and the commit and taking care of this for me. Also, thanks @HBelusca and @JoachimHenze for your reviews, comments, and approvals. The credit for the idea here goes to @ThFabba.

**Pull request merged**